tests/feature/Rules/KeyOptionalTest.php

```diff
@@ -9,70 +9,70 @@
 
 test('Default', expectAll(
     fn() => v::keyOptional('foo', v::intType())->assert(['foo' => 'string']),
-    'foo must be an integer',
-    '- foo must be an integer',
-    ['foo' => 'foo must be an integer']
+    '`.foo` must be an integer',
+    '- `.foo` must be an integer',
+    ['foo' => '`.foo` must be an integer'],
 ));
 
 test('Inverted', expectAll(
     fn() => v::not(v::keyOptional('foo', v::intType()))->assert(['foo' => 12]),
-    'foo must not be an integer',
-    '- foo must not be an integer',
-    ['foo' => 'foo must not be an integer']
+    '`.foo` must not be an integer',
+    '- `.foo` must not be an integer',
+    ['foo' => '`.foo` must not be an integer'],
 ));
 
 test('Inverted with missing key', expectAll(
     fn() => v::not(v::keyOptional('foo', v::intType()))->assert([]),
-    'foo must be present',
-    '- foo must be present',
-    ['foo' => 'foo must be present']
+    '`.foo` must be present',
+    '- `.foo` must be present',
+    ['foo' => '`.foo` must be present'],
 ));
 
 test('With wrapped name, default', expectAll(
     fn() => v::keyOptional('foo', v::intType()->setName('Wrapped'))->setName('Wrapper')->assert(['foo' => 'string']),
     'Wrapped must be an integer',
     '- Wrapped must be an integer',
-    ['foo' => 'Wrapped must be an integer']
+    ['foo' => 'Wrapped must be an integer'],
 ));
 
 test('With wrapped name, inverted', expectAll(
     fn() => v::not(v::keyOptional('foo', v::intType()->setName('Wrapped'))->setName('Wrapper'))->setName('Not')->assert(['foo' => 12]),
     'Wrapped must not be an integer',
     '- Wrapped must not be an integer',
-    ['foo' => 'Wrapped must not be an integer']
+    ['foo' => 'Wrapped must not be an integer'],
 ));
 
 test('With wrapper name, default', expectAll(
     fn() => v::keyOptional('foo', v::intType())->setName('Wrapper')->assert(['foo' => 'string']),
-    'foo must be an integer',
-    '- foo must be an integer',
-    ['foo' => 'foo must be an integer']
+    'Wrapper must be an integer',
+    '- Wrapper must be an integer',
+    ['foo' => 'Wrapper must be an integer'],
 ));
 
 test('With wrapper name, inverted', expectAll(
     fn() => v::not(v::keyOptional('foo', v::intType())->setName('Wrapper'))->setName('Not')->assert(['foo' => 12]),
-    'foo must not be an integer',
-    '- foo must not be an integer',
-    ['foo' => 'foo must not be an integer']
+    'Wrapper must not be an integer',
+    '- Wrapper must not be an integer',
+    ['foo' => 'Wrapper must not be an integer'],
 ));
 
 test('With "Not" name, inverted', expectAll(
     fn() => v::not(v::keyOptional('foo', v::intType()))->setName('Not')->assert(['foo' => 12]),
-    'foo must not be an integer',
-    '- foo must not be an integer',
-    ['foo' => 'foo must not be an integer']
+    'Not must not be an integer',
+    '- Not must not be an integer',
+    ['foo' => 'Not must not be an integer'],
```

Comment:
The above 3 changes are a massive bonus, but they could also be a liability. 

The problem here is if you have a series of children, they will all have the same name. This is not desirable.

Reply:
Even though they're not related, we should not overwrite the name of rules that have paths. Rules that have paths should have a name deliberately set. A way to go about it is probably not to update the name of children that have paths, but update the name of the parent is all cases.